lib/pagseguro/transaction/response.rb

```diff
@@ -1,11 +1,44 @@
 module PagSeguro
   class Transaction
     class Response
-      # Set the response errors.
-      attr_reader :errors
+      def initialize(response, object)
+        @response = response
+        @object = object
+      end
+
+      def serialize
+        if success?
+          object.update_attributes(Serializer.new(xml).serialize)
+        else
+          object.errors.add response
+        end
+
+        object
+      end
+
+      def serialize_collection
+        if success?
+          object.transactions = Serializer.new(xml).serialize_status_history
```

Comment:
The `serialize_status_history` method already iterates over all transactions from xml?